### DISCUSSION TITLE
Point mass to vary CG

### DISCUSSION THREAD

**Discussion by gasilva** [JSBSim-Team/jsbsim#482] [Q&A]
@seanmcleod see if this way of varying CG is better than changing the aircraft XML

```python
# Developed by JSBSim Team and
# Guilherme A. L. da Silva - aerothermalsolutions.co
# GNU Lesser General Public License v2.1

# Varies point mass weigh and location to vary CG

# To install jsbsim module in Python
# pip install jsbsim
# Or conda install jsbsim (if you have anaconda, this is the best way)

import jsbsim
import os

# Put your path here
PATH_TO_JSBSIM_FILES="/Users/gasilva/Documents/GitHub/jsbsimNew" #my is a macbook
# Object with JSB executable
fdm = jsbsim.FGFDMExec(PATH_TO_JSBSIM_FILES)
# Load script

fdm.load_model('global5000')

fdm['propulsion/engine[0]/set-running'] = 1
fdm['propulsion/engine[1]/set-running'] = 1
fdm['ic/h-sl-ft'] = 30000
fdm['ic/gamma-deg'] = 0
fdm['ap/alpha_hold']= 0
fdm['ap/altitude_hold']= 0
fdm['ic/vc-kts']=280

# Run JSBSim
fdm.run_ic()
fdm.run()

cgx=fdm["inertia/cg-x-in"]
print("------------------------------")
print("Original X location {:.1f} in".format(cgx))
print("------------------------------")

newx=cgx*2
neww=1000
fdm["inertia/pointmass-location-X-inches"]=newx
fdm["inertia/pointmass-weight-lbs"]=neww

fdm.run()
print("X location {:.1f} in".format(newx))
print("point mass {:.0f} in".format(neww))
print("New CG {:+0.2f} %".format((fdm["inertia/cg-x-in"]/cgx-1)*100))
print("------------------------------")

newx=cgx*.5
neww=2000
fdm["inertia/pointmass-location-X-inches"]=newx
fdm["inertia/pointmass-weight-lbs"]=neww

fdm.run()

print("X location {:.1f} in".format(newx))
print("point mass {:.0f} in".format(neww))
print("New CG {:+0.2f} %".format((fdm["inertia/cg-x-in"]/cgx-1)*100))
print("------------------------------")
```
Results example

```bash
------------------------------
Original X location 790.8 in
------------------------------
X location 1581.6 in
point mass 1000 lb
New CG +1.36 %
------------------------------
X location 395.4 in
point mass 2000 lb
New CG -1.34 %
------------------------------
```

**Comment by seanmcleod** [comment]
Yes this is what I was thinking of. If you take a look you'll see `inertia/iyy-slugs_ft2` also changing as you move the point mass.

**Reply by gasilva** [reply]
It is really more realistic doing this way.

**Comment by seanmcleod** [comment]
First prize would be configuring various point masses, fuel tanks included that match the actual layout of the various masses loaded in the aircraft in order to achieve your mass target and your cg target. As opposed to a single point mass in your example above.

For example if you take a look at the 737 model in JSBSim you'll see that there are 3 fuel tanks specified at different locations.

https://github.com/JSBSim-Team/jsbsim/blob/24f41de4a77ef8571b57dd125042554e6aae7471/aircraft/737/737.xml#L167-L196

So if your target mass was say `TM`, and the empty mass is say `EM`, then you would load in `TM - EM` of fuel, and partition it between the fuel tanks to achieve your target `cg`.

What is a bit suspicious about the Global 5000 aircraft is that all 3 fuel tanks specified coincide exactly with the aircraft's empty cg, here is the empty cg. 

https://github.com/JSBSim-Team/jsbsim/blob/24f41de4a77ef8571b57dd125042554e6aae7471/aircraft/global5000/global5000.xml#L87-L91

And here are the 3 fuel tanks.

https://github.com/JSBSim-Team/jsbsim/blob/24f41de4a77ef8571b57dd125042554e6aae7471/aircraft/global5000/global5000.xml#L235-L263

For additional realism and flexibility take a look at the following 737-800 model, which in addition to specifying 3 fuel tank locations they also include a number of cabin areas and cargo areas as point masses.

https://github.com/zakalawe/737-800/blob/d1f2b32031ea7c7993ac3c8ed994ef93b4e8886f/738.xml#L43-L139
